drivers/rilmodem/voicecall.c

```diff
@@ -598,10 +599,18 @@ static void ril_ss_notify(struct ril_msg *message, gpointer user_data)
 
 void ril_answer(struct ofono_voicecall *vc, ofono_voicecall_cb_t cb, void *data)
 {
+	struct ril_voicecall_data *vd = ofono_voicecall_get_data(vc);
+	struct parcel rilp;
+
 	DBG("Answering current call");
 
+	g_ril_request_answer(vd->ril, &rilp);
+
 	/* Send request to RIL */
-	ril_template(RIL_REQUEST_ANSWER, vc, generic_cb, 0, NULL, cb, data);
+	if (rilp.size > 0)
+		ril_template(RIL_REQUEST_ANSWER, vc, generic_cb, 0, &rilp, cb, data);
+	else
+		ril_template(RIL_REQUEST_ANSWER, vc, generic_cb, 0, NULL, cb, data);
```

Comment:
I guess this can be re-written to:
```suggestion
	ril_template(RIL_REQUEST_ANSWER, vc, generic_cb, 0,
				rilp.size > 0 ? &rilp : NULL, cb, data);
```